packages/core/src/components/items/_types/IMenuItemNameProps.ts

```diff
@@ -0,0 +1,17 @@
+import {ISearchHighlighterProps} from "../../../actions/types/search/_types/ISearchHighlighterProps";
+import {ISimpleSearchPatternMatcher} from "../../../actions/types/search/_types/ISimpleSearchPatternMatcher";
+import {IQuery} from "../../../menus/menu/_types/IQuery";
+import {ISubscribable} from "../../../utils/subscribables/_types/ISubscribable";
+import {LFC} from "../../../_types/LFC";
+
+/** The props for the standard menu item name */
+export type IMenuItemNameProps = {
+    /** The textual string to show */
+    name: ISubscribable<string>;
+    /** The pattern matcher to use to extract the text to highlight from the query (E.g. removes prefix) */
+    searchPattern?: ISimpleSearchPatternMatcher;
+    /** The query to highlight in the description */
+    query?: IQuery | null;
```

Comment:
Is there a reasong why you allow `null` here and not just `undefined`? If so might want to document it.

Reply:
Not really, they serve the same purpose. But undefined usually acts as just not specifying something, where as null indicates you explicitly passed it. I'm not very consistent at it, but the rule of thumb should be to not pass `undefined` in your code at all, but have that be a result of something really not being present. And here I want to allow for both implicit and explicitly absence of a query. (Besides, this type just mirrors the data that could be passed to the component where it's used)

Reply:
Fair if it is just to deminstrate intent, that's fair.